streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -1321,11 +1322,22 @@ int process(final int maxNumRecords, final Time time) {
                 totalProcessed += processed;
                 task.recordProcessBatchTime(now - then);
             }
+            successfullyProcessed.add(task);
         }
-
+        successfullyProcessed.clear();
```

Comment:
nit: if we just clear it at the beginning of `#process` then we only need to clear it in one place, vs clearing it here and also in `#commitSuccessfullyProcessedTasks`